plugin/execute_command.py

```diff
@@ -0,0 +1,36 @@
+from .core.registry import client_for_view, LspTextCommand
+from .core.settings import client_configs
+from .core.protocol import Request
+
+try:
+    from typing import List, Optional, Dict, Any
+    assert List and Optional and Dict and Any
+except ImportError:
+    pass
+
+
+class LspExecuteCommand(LspTextCommand):
+    def __init__(self, view):
+        super().__init__(view)
+
+    def run(self, lsp_command) -> None:
+        self._commands = []   # type: List[str]
+        for config in client_configs.all:
+            if config.commands:
+                self._commands.extend(config.commands)
+
+        if len(self._commands) > 0:
+            self.view.window().show_quick_panel(self._commands, self._on_done)
+
+    def _handle_response(self, response: 'Optional[Any]') -> None:
+        pass
+
+    def _on_done(self, index: int) -> None:
+        if index > -1:
+            command = self._commands[index]
+            client = client_for_view(self.view)
+            if client:
+                request = {
+                    "command": command
+                }
```

Comment:
How are you planning to use string arguments?

Reply:
yes I am planning to forward the command args as well

Reply:
BTW according to the spec the command type is a string
https://microsoft.github.io/language-server-protocol/specification#workspace_executeCommand

Reply:
> BTW according to the spec the command type is a string

I'm not sure I totally follow you here. The `arguments` is an `any`, so can potentially be a list or something. My question was: how do we know if the command should supply a string argument, or a null argument? I think it's good to restrict us to no arguments and a single string argument for now.

Reply:
ouuhh sorry totally miss read your initial question. 

> I'm not sure I totally follow you here. The `arguments` is an `any`, so can potentially be a list or something. My question was: how do we know if the command should supply a string argument, or a null argument? I think it's good to restrict us to no arguments and a single string argument for now.

I am fine with treating the arguments for now as an optional single one of type string. But assuming we want to be more flexible, is there a problem to read the arguments as Dict[String, Any] ? 

@olafurpg is any of the current/future Metals commands need arguments ?

Reply:
From the doc, Metals commands don't take arguments for now
https://scalameta.org/metals/docs/editors/new-editor.html#metals-server-commands

Reply:
I don't have any planned commands in mind for Metals that will need arguments or return meaningful responses (they all return `{}`).